src/java.base/share/classes/java/net/ServerSocket.java

```diff
@@ -929,7 +924,17 @@ public String toString() {
      *             {@code checkSetFactory} method doesn't allow the operation.
      * @see        java.net.SocketImplFactory#createSocketImpl()
      * @see        SecurityManager#checkSetFactory
+     * @deprecated Use {@link ServerSocketChannel}, or subclass {@code ServerSocket}
+     *    directly.
```

Comment:
A better suggestion in the first line would be:

```
Use a {@link ServerSocketFactory} and subclass {@code ServerSocket} directly.
```

Reply:
I've incorporated your suggestion now. See commit 3a89432